src/ExpressiveInstaller/Resources/src/ExpressiveSymfonyContainer.php

```diff
@@ -0,0 +1,192 @@
+<?php
+
+namespace App;
+
+use Symfony\Component\DependencyInjection\Container;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Dumper\PhpDumper;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * Configuration for the Symfony Dependency Injection container.
+ *
+ * This class provides functionality for the following service types:
+ *
+ * - Aliases
+ * - Delegators
+ * - Factories
+ * - Invokable classes
+ * - Services (known instances)
+ */
+class ExpressiveSymfonyContainer
+{
+    const ENABLE_CACHE = 'cache_symfony_container';
+
+    /**
+     * @var array
+     */
+    private $config;
+
+    /**
+     * @var string
+     */
+    private $cachedContainerFile;
+
+    /**
+     * @param array $config
+     * @param string $cachedContainerFile file name to use to store the cached container
+     */
+    public function __construct(
+        array $config,
+        $cachedContainerFile
+    ) {
+        $this->config = $config;
+        $this->cachedContainerFile = $cachedContainerFile;
+    }
+
+    /**
+     * @return Container
+     */
+    public function create()
+    {
+        $cacheEnabled = file_exists($this->cachedContainerFile) &&
+            isset($this->config[static::ENABLE_CACHE]) &&
+            $this->config[static::ENABLE_CACHE];
+
+        if (! $cacheEnabled) {
+            $this->cacheContainer($this->build());
+        }
+
+        $container = $this->loadFromCache($this->cachedContainerFile);
+
+        // Inject config
+        $container->set('config', $this->config);
```

Comment:
No, the `dependencies` key is only for configuring the container.

This is for zend-servicemanager.
```php
// Load configuration
$config = require __DIR__ . '/config.php';

// Build container
$container = new ServiceManager();
(new Config($config['dependencies']))->configureServiceManager($container);

// Inject config
$container->setService('config', $config);
```
It loads the config, grabs the dependencies to build the container and then injects the config so factories can access it. I can't think of any reason why someone would need access to the `dependencies` key.

Besides the dependencies the complete config for the application is in that `$config` array. An example of a [template config](https://github.com/xtreamwayz/xtreamwayz.com/blob/master/config/autoload/templates.global.php).

Some other containers don't accept the `$config` as an array and they convert it to an ArrayObject:
```php
new ArrayObject($this->config, ArrayObject::ARRAY_AS_PROPS)```